### ISSUE TITLE
Intended way to disable file persistence

### ISSUE THREAD

**Issue by cdealti** [moquette-io/moquette#242]
The only way I've found to disable the file persistence and use the in memory DB is to append a space after ``persistent_store`` in the configuration file

```
# To disable persistence add one trailing space :-(
persistent_store 
```

Removing the trailing space throws:

```
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(String.java:1955)
	at io.moquette.server.config.ConfigurationParser.parse(ConfigurationParser.java:109)
	at io.moquette.server.config.ConfigurationParser.parse(ConfigurationParser.java:69)
	at io.moquette.server.config.FilesystemConfig.<init>(FilesystemConfig.java:38)
	at io.moquette.server.config.FilesystemConfig.<init>(FilesystemConfig.java:46)
	at io.moquette.server.Server.startServer(Server.java:71)
	at io.moquette.server.Server.main(Server.java:55)
	at org.eclipse.kura.core.test.AllCoreTests.setUpClass(AllCoreTests.java:78)
```

**Comment by andsel** [comment]
Hi Cristiano
the guilty is the method assignDefaults() invoked int the FilesystemConfig (but that method is invoked also in the other types of config, fro example MemoryConfig) constructor and the rationale was "provide the defaults value to every config to let the broker starts", and the path of the mapdb file is a mandatory property to start the broker.
 But what you can do to start the broker in embedded mode is to provide your self a set of properties (using the MemoryConfig using the constructor that let you specify you properties), specially redefine the BrokerConstants.PERSISTENT_STORE_PROPERTY_NAME to empty string.  The logic behind is that the user provided properties overrides the default ones. 

 To see an example of how to embedded correctly Moquette please refer to EmbeddedLauncher, 


 Andera

**Comment by cdealti** [comment]
I wonder if an in memory persistence configuration would be a more appropriate default (like for Mosquitto if I'm not wrong).

I'm embedding Moquette in the Eclipse Kura tests and the Moquette configuration file is something I'd like to continue using.

**Comment by andsel** [comment]
The logic behind default proerty creation for MapDB files could be inverted, so the default settings could become "use Memory", and else if a path is defined use that.

**Comment by cdealti** [comment]
seems reasonable to me

**Comment by andsel** [comment]
Fixed with 0f41c9ea0853633ad3d534e8113a3df75bf6bff5

**Comment by andsel** [comment]
Hi @cdealti how is going with this patch it solved and served to include Moquette  in the Kura's tests?
